cdk/src/open-data-platform/data-plane/schema/schema.handler.ts

```diff
@@ -108,6 +100,23 @@ async function runSchemaFile(file: string, db: ConnectionPool) {
   return await db.query(sql.file(file));
 }
 
+async function createDatabaseConfig(): Promise<ConnectionPoolConfig> {
+  console.log('Fetching db credentials...');
+  const { host, port, username, password } = await getCredentials(CREDENTIALS_SECRET);
+  return {
+    host: host,
+    port: port,
+    user: username,
+    password: password,
+  };
+}
+
+export async function connectToDb(config?: ConnectionPoolConfig): Promise<ConnectionPool> {
```

Comment:
Why does this function exist when we have `database` already? Is there a case when `config` can be null?

Reply:
Since we don't want the handlers to have to do this credential fetching themselves (it's the same every time), it's null. Happy to update this with any suggestions you may have

Reply:
When does the credential get fetched if not during the handler execution?

Reply:
Through createDatabaseConfig(). Should I not even allow the option to pass a config?

Reply:
Does that get called whenever `handler` runs? Could you walk me through how this code means we don't have to fetch the credentials every time?

Reply:
I added some comments to explain what's happening. The data writing handlers (demographics and water systems) call connectToDb() so that they don't repeat the same getCredential and database() code. The schema handler's handle() function also needs access to the db config so that is its own helper as well